src/viewport.cpp

```diff
@@ -1478,14 +1547,10 @@ void ViewportSign::MarkDirty(ZoomLevel maxzoom) const
 		zoomlevels[zoom].bottom = this->top    + ScaleByZoom(VPSM_TOP + FONT_HEIGHT_NORMAL + VPSM_BOTTOM + 1, zoom);
 	}
 
-	Window *w;
-	FOR_ALL_WINDOWS_FROM_BACK(w) {
-		ViewPort *vp = w->viewport;
-		if (vp != nullptr && vp->zoom <= maxzoom) {
-			assert(vp->width != 0);
-			Rect &zl = zoomlevels[vp->zoom];
-			MarkViewportDirty(vp, zl.left, zl.top, zl.right, zl.bottom);
-		}
+	for (Viewport *vp : ViewportData::cache) {
+		if ( vp->zoom > maxzoom) continue;
```

Comment:
extra space :p

Reply:
I must have missed that when I refactored my refactor. Also, I'm getting tired of staring at the "FIXME" in the line just above it. What can I do about that? :thinking: